Frameworks/Security/GenericPasswordItemHandler.mm

```diff
@@ -589,16 +558,14 @@ - (OSStatus)remove:(NSDictionary*)queryDictionary {
         return errSecParam;
     }
 
-    // TODO: Error handling when projections support it.
-    NSArray* credArray = [_vault retrieveAll];
+    WFC::IVectorView<PasswordCredential> credArray = _vault.RetrieveAll();
 
     // Documentation for remove states that it by default removes all matching rows while documentation
     // for query defaults to a single item. Go figure.
-    std::vector<StrongId<WSCPasswordCredentialWrapper>> credentialsToRemove = FindMatchingCredentials(queryDictionary, credArray, true);
+    std::vector<StrongId<PasswordCredentialWrapper>> credentialsToRemove = FindMatchingCredentials(queryDictionary, credArray, true);
 
     for (const auto& credential : credentialsToRemove) {
-        // TODO: Error handling when projections support it.
```

Comment:
The comment dated from a time when the projections swallowed errors instead of throwing them. I removed it because it was out of date.

C++/WinRT handles errors in pretty much the same way that our projections do: by throwing them. Most WinRT APIs should not throw under normal circumstances, so I wouldn't advocate putting catch blocks everywhere. Maybe we should at least translate the C++ exceptions to Objective-C exceptions?